### PR TITLE
Add endpoint implementation of Instrument

### PR DESCRIPTION
**Pull request by lerenn**
Hi !

As you've done a great job with the start of this API, I contributed for the part I needed.
If there is anything to change, just ask ;)

### CODE REVIEW

File: instrument_test.go [hunk 1]
```diff
@@ -0,0 +1 @@
+package oanda
```
Comment (nii236) [inline instrument_test.go]: No need for this file (delete)

File: models.go [hunk 2]
```diff
@@ -46,6 +46,29 @@ type Account struct {
 	LastTransactionID           string `json:"lastTransactionID"`
 }
 
+// CandlesContainer contains a slice of Candle
+type CandlesContainer struct {
+	Candles []Candle
+}
+
+// Candle contains informations concerning a candle
```
Comment (nii236) [inline models.go]: > Candle

Instead of 

> candle

File: instrument.go [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+	CandlesTo            time.Time
+	CandlesSmooth        bool
+	CandlesIncludeFirst  bool
+	CandlesDailyAlign    int
+	CandlesAlignTimezone time.Location
+	CandlesWeeklyAlign   time.Weekday
+)
+
+// Private
+////////////////////////////////////////////////////////////////////////////////
```
Comment (nii236) [inline instrument.go]: Please remove this line

File: instrument.go [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+	err = json.Unmarshal(b, &result)
+	if err != nil {
+		return nil, errors.Wrap(err, "Could not unmarshal JSON")
+	}
+
+	return &result, nil
+}
+
+// Arguments
+////////////////////////////////////////////////////////////////////////////////
```
Comment (nii236) [inline instrument.go]: Please remove this line

File: examples/instruments.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+	container, err := c.GetCandles("EUR_GBP")
+
+	// Display
+	nbr := len(container.Candles)
+	fmt.Println("No options:")
+	fmt.Println(0, container.Candles[0])
+	fmt.Println(nbr, container.Candles[nbr-1])
+
+	// Test with all options
+	loc, _ := time.LoadLocation("Europe/Paris")
```
Comment (nii236) [inline examples/instruments.go]: Please check this error

File: examples/instruments.go [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+		"EUR_USD",
+		oanda.CandlesAlignTimezone(*loc),
+		oanda.CandlesCount(2),
+		oanda.CandlesDailyAlign(2),
+		oanda.CandlesFrom(time.Now().AddDate(0, -15, 0)),
+		oanda.CandlesGranularity("D"),
+		oanda.CandlesIncludeFirst(false),
+		oanda.CandlesPrice("BA"),
+		oanda.CandlesSmooth(true),
+		//oanda.CandlesTo(time.Now().AddDate(0, -1, 0)), -> You can't set From and To at the same time
```
Comment (nii236) [inline examples/instruments.go]: Please remove commented code

File: instrument.go [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+	CandlesIncludeFirst  bool
+	CandlesDailyAlign    int
+	CandlesAlignTimezone time.Location
+	CandlesWeeklyAlign   time.Weekday
+)
+
+// Private
+////////////////////////////////////////////////////////////////////////////////
+
+func (p CandlesPrice) applyCandlesArg(v *url.Values) {
```
Comment (nii236) [inline instrument.go]: Prefer `cp` for the struct receiver

File: instrument.go [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+	v.Set("price", string(p))
+}
+
+func (g CandlesGranularity) applyCandlesArg(v *url.Values) {
```
Comment (nii236) [inline instrument.go]: Prefer `cg` for the struct receiver

File: instrument.go [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+	v.Set("granularity", string(g))
+}
+
+func (c CandlesCount) applyCandlesArg(v *url.Values) {
```
Comment (nii236) [inline instrument.go]: Prefer `cc` for the struct receiver

File: instrument.go [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+	v.Set("count", strconv.Itoa(int(c)))
+}
+
+func (f CandlesFrom) applyCandlesArg(v *url.Values) {
```
Comment (nii236) [inline instrument.go]: Prefer `cf` for the struct receiver

File: instrument.go [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+	tf := time.Time(f).Format(candlesTimeFormat)
+	v.Set("from", tf)
+}
+
+func (t CandlesTo) applyCandlesArg(v *url.Values) {
```
Comment (nii236) [inline instrument.go]: Prefer `ct` for the struct receiver

File: instrument.go [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+	tf := time.Time(t).Format(candlesTimeFormat)
+	v.Set("to", tf)
+}
+
+func (s CandlesSmooth) applyCandlesArg(v *url.Values) {
```
Comment (nii236) [inline instrument.go]: Prefer `cs` for the struct receiver

File: instrument.go [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+	v.Set("smooth", strconv.FormatBool(bool(s)))
+}
+
+func (i CandlesIncludeFirst) applyCandlesArg(v *url.Values) {
```
Comment (nii236) [inline instrument.go]: Prefer `cif` for the struct receiver

File: instrument.go [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+	v.Set("includeFirst", strconv.FormatBool(bool(i)))
+}
+
+func (da CandlesDailyAlign) applyCandlesArg(v *url.Values) {
```
Comment (nii236) [inline instrument.go]: Prefer `cda` for the struct receiver

File: instrument.go [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+	v.Set("dailyAlignment", strconv.Itoa(int(da)))
+}
+
+func (at CandlesAlignTimezone) applyCandlesArg(v *url.Values) {
```
Comment (nii236) [inline instrument.go]: Prefer `cat` for the struct receiver

File: instrument.go [hunk 16]
```diff
@@ -0,0 +1,136 @@
...
+	loc := time.Location(at)
+	v.Set("alignmentTimezone", loc.String())
+}
+
+func (wa CandlesWeeklyAlign) applyCandlesArg(v *url.Values) {
```
Comment (nii236) [inline instrument.go]: Prefer `cwa` for the struct receiver

**Review by nii236 — CHANGES_REQUESTED** [review]

**Comment by nii236** [comment]
Thanks so much for your contribution! Please make the minor changes above and I'll be happy to merge it in 👍

**Comment by lerenn** [comment]
Done ! It should be as you expected it.

**Comment by nii236** [comment]
Awesome thanks for your contribution!

**Pull request merged**